api_app/analyzers_manager/observable_analyzers/dns0_names.py

```diff
@@ -0,0 +1,212 @@
+        ):
+            raise AnalyzerConfigurationException(
+                f"{self.limit} is out of bound! "
+                f"Max value is {_max_limit_value}, min value is {_min_limit_value}"
+            )
+
+        if hasattr(self, "offset") and self.offset < _min_offset_value:
+            raise AnalyzerConfigurationException(
+                f"{self.offset} can't be below {_min_offset_value}"
+            )
+
+    @staticmethod
+    def convert_date_type(date_string):
+        if not date_string:
+            return False
+
+        return (
+            DNS0Names.convert_unix_timestamp(date_string)
+            or DNS0Names.convert_relative_date(date_string)
+            or DNS0Names.convert_date(date_string)
+        )
+
+    @staticmethod
+    def convert_relative_date(date):
+        # accepts string matching the format:
+        # - at the beginning
+        # a number
+        # a character indicating Year, Month or Day
+        pattern = re.compile(r"-\d+[YMD]")
+        if match := pattern.match(date):
+            return match.group()
+        return False
+
+    @staticmethod
+    def convert_date(date):
+        try:
+            parsed = dateutil_parser.parse(date).strftime("%Y-%m-%d")
+            return parsed
+        except Exception:
+            return False
+
+    @staticmethod
+    def convert_unix_timestamp(timestamp):
+        try:
+            parsed = str(int(timestamp))
+            return parsed
+        except Exception:
+            return False
+
+    def _create_params(self):
+        target_observable = self.observable_name
+        if self.observable_classification == self.ObservableTypes.URL:
+            target_observable = urlparse(self.observable_name).hostname
+        params = {"q": target_observable}
+
+        # convert root parameter into 1 or 0
+        if hasattr(self, "root") and self.root:
+            params["root"] = int(self.root)
+
+        # pass list of fuzzy parameter
+        if hasattr(self, "fuzzy") and self.fuzzy:
+            params["fuzzy"] = self.fuzzy
+
+        # convert dates to correct format
+        if hasattr(self, "from_date") and self.from_date:
+            if result := self.convert_date_type(self.from_date):
+                params["from"] = result
+
+        if hasattr(self, "to_date") and self.to_date:
+            if result := self.convert_date_type(self.to_date):
+                params["to"] = result
+
+        if hasattr(self, "not_before") and self.not_before:
+            if result := self.convert_date_type(self.not_before):
+                params["not_before"] = result
+
+        if hasattr(self, "sort") and self.sort:
+            params["sort"] = self.sort
+
+        if hasattr(self, "format") and self.format:
+            params["format"] = self.format
+
+        if hasattr(self, "limit") and self.limit:
+            params["limit"] = self.limit
+
+        if hasattr(self, "offset") and self.offset:
+            params["offset"] = self.offset
```

Comment:
maybe you can try to avoid replication of this `if` clauses by optimizing this code and looping for the attributes